### ISSUE TITLE
Indeterministic behavior of esmini

### ISSUE THREAD

**Issue by huber-florian** [esmini/esmini#449]
Is there a possibility to start esmini from Python code with --fixed_timestep 0.02 ?
Currently, I start it with "se.SE_Init(input_as_bytes, 0, 1, 0, 0)".

Unfortunately, I have many problems with indeterministic behavior of esmini currently. Should the behavior be completely deterministic afterwards?

**Comment by eknabevcc** [comment]
First of all. For deterministic behavior, use `SE_StepDT(float dt)`. If you run the same scenario twice, with the same dt, you will get the same result.

The use case for SE_Step() is to make esmini choose timestep based on actual elapsed time - in order to synchronize with actual real time, so that the scenario runs and appears at the intended speed. (it can even cause esmini to fall asleep between steps if elapsed time is very small, to preserve CPU load)

Regarding determinism, there are a few special cases to be aware of:

1. Choice route in junctions/intersections 
   If no route is defined for an entity, the choice is random. To ensure determinism, either specify seed (`SE_SetSeed(unsigned int seed)`) or, even better, assign a route to the entity, ensuring it will always take the same decision.
2. [TrafficSwarmAction](https://www.asam.net/static_downloads/ASAM_OpenSCENARIO_V1.2.0_Model_Documentation/modelDocumentation/content/TrafficSwarmAction.html). This action will populate stochastic traffic around a specified central entity. Also in this case, specifying the seed will make swarm exactly traffic the same each run.

In summary, to achieve full repeatability:
1. Use SE_StepDT(dt) witch same dt
2. Specify seed and specify routes for entities
(second step only relevant when involving intersections and/or swarm traffic)

Please try this and report back if it helps eliminate the indeterministic behavior or not. If not, we should of course investigate further.

**Comment by huber-florian** [comment]
Thank you! SE_StepDT(float dt) already solved all my problems.
